### ISSUE TITLE
Shouldn't `flatMap`ing to `observable.take(1)` behave like `map(observable)`?

### ISSUE THREAD

**Issue by timmolendijk** [baconjs/bacon.js#537]
Take a look at [this demo](http://jsbin.com/fivuwanihu/3/edit?html,js,output). Every time you click "celebrate birthday" our (behind-the-scenes) age counter should move up by one, and every time you click "log current age" it should publish the current age. Also: when you click the first button no log should appear, and when you click the second the age should not change.

As you can witness it does not behave correctly. The problem seems to be that `ageProperty` never gets subscribed to and remains dormant indefinitely. Why not? I had expected the `flatMap` to behave identical to `map(ageProperty)`.

The funny thing is that this is exactly what happens, _iff_ you make sure `ageProperty` has been subscribed to before it gets `flatMap`'d.

Another funny thing is that as soon as you get rid of the `.take(1)` it _does_ get subscribed to as a result of the `flatMap` and everything behaves as expected (but differently obviously).

Although I cannot really explain the exact nature of this issue, it does smell a lot like the issue I addressed last week on some fundamental problems with lazy properties (#536). Isn't this yet another piece of evidence that _lazy evaluation_ is irreconcilable with the notion of _having a value_?

**Comment by phadej** [comment]
It behaves in the right way.

> The problem seems to be that ageProperty never gets subscribed to and remains dormant indefinitely. Why not?

Because of dynamic nature of `flatMap`, we can't never know what's coming next in flatMapped array.
`ageProperty.take(1)` returns a value, not a stream.

So if you don't explicitly subscribe, lazy propagation eats age events.

With `clicks.map(ageProperty)` (which is `ageProperty.sampledBy(clicks)`) the dependency is clear, so `ageProperty` is subscribed onto.
Also in `clicks.flatMap(() => ageProperty)` `ageProperty` is subscribed onto. Actually many times, once per each click. 

---

Bottom line: *Yes*, this is related to lazy propagation. But currently works by design.

**Comment by timmolendijk** [comment]
@phadej:
> `ageProperty.take(1)` returns a value, not a stream.

How do you mean? I'm pretty sure `ageProperty.take(1) instanceof Bacon.Property === true`. Which is why it is very hard to make sense of why flatmapping `ageProperty` _does_ work while flatmapping `ageProperty.take(1)` doesn't.

**Comment by raimohanska** [comment]
I'm totally late to the party again :( Anyways, @phadej's suggestion is correct, you should sample the age property instead of flatmapping, because the latter approach only subscribes to the property when sampling, and resulting to the property not being kept up to date.

And why `flatmap -> ageProperty` works even though `flatmap -> ageProperty.take(1)` doesnt? That's because this way the subscription to `ageProperty` is left active instead of being broken on each sample.
